packages/block-editor/src/components/block-navigation/tree.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+
+import { __experimentalTreeGrid as TreeGrid } from '@wordpress/components';
+import { useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import BlockNavigationBranch from './branch';
+import { BlockNavigationContext } from './context';
+
+/**
+ * Wrap `BlockNavigationRows` with `TreeGrid`. BlockNavigationRows is a
+ * recursive component (it renders itself), so this ensures TreeGrid is only
+ * present at the very top of the navigation grid.
+ *
+ * @param {Object} props
+ */
+export default function BlockNavigationTree( {
```

Comment:
@adamziel I renamed `BlockNavigationList` to `BlockNavigationTree` to be consistent with the whole tree/branch/leaf thing.